packages/core/src/getStateFromPath.tsx

```diff
@@ -254,3 +272,56 @@ function findParseConfigForRoute(
   }
   return undefined;
 }
+
+function findInitialRoute(
+  routeName: string,
+  initialRoutes: InitialRouteConfig[]
+): string | undefined {
+  for (const config of initialRoutes) {
+    if (config.connectedRoutes.includes(routeName)) {
+      return config.initialRouteName === routeName
+        ? undefined
+        : config.initialRouteName;
+    }
+  }
+  return undefined;
+}
+
+function parseRoute(
+  state: InitialState,
+  initialRoute: string | undefined,
+  routeName: string,
+  isEmpty: boolean,
+  params?: Record<string, any> | undefined
+) {
+  if (isEmpty) {
```

Comment:
looks good, but again, make it please more descriptive. Also, could you try to omit to make side effects and use `return`?

Reply:
can do